src/gui/updater/appimageupdater.cpp

```diff
@@ -0,0 +1,202 @@
+#include "application.h"
+
+using namespace OCC;
+using namespace std::chrono_literals;
+
+namespace {
+
+/**
+ * libappimageupdate uses exceptions, but the client does not
+ * This little shim adapts the interface to one usable within this project
+ */
+class AppImageUpdaterShim : public QObject
+{
+    Q_OBJECT
+
+private:
+    explicit AppImageUpdaterShim(const QString &zsyncFileUrl, QObject *parent = nullptr)
+        : QObject(parent)
+        , _updater(Utility::appImageLocation().toStdString(), true)
+    {
+        QString updateInformation(QStringLiteral("zsync|") + zsyncFileUrl);
+        _updater.setUpdateInformation(updateInformation.toStdString());
+    }
+
+    void _logStatusMessages()
+    {
+        std::string currentStatusMessage;
+
+        while (_updater.nextStatusMessage(currentStatusMessage)) {
+            qCInfo(lcUpdater) << "AppImageUpdate:" << QString::fromStdString(currentStatusMessage);
+        }
+    }
+
+public:
+    static AppImageUpdaterShim *makeInstance(const QString &updateInformation, QObject *parent)
+    {
+        try {
+            return new AppImageUpdaterShim(updateInformation, parent);
+        } catch (const std::exception &e) {
+            qCCritical(lcUpdater) << "Failed to create updater shim:" << e.what();
+            return nullptr;
+        }
+    }
+
+    bool isUpdateAvailable() noexcept
+    {
+        try {
+            bool updateAvailable;
+
+            if (!_updater.checkForChanges(updateAvailable)) {
+                _logStatusMessages();
+                return false;
+            }
+
+            _logStatusMessages();
+            return updateAvailable;
+        } catch (const std::exception &e) {
+            _logStatusMessages();
+            qCCritical(lcUpdater) << "Checking for update failed:" << e.what();
+            return false;
+        }
+    }
+
+    void startUpdateInBackground() noexcept
+    {
+        // monitor progress and log status messages
+        auto *timer = new QTimer(this);
+
+        timer->setInterval(100ms);
+
+        connect(timer, &QTimer::timeout, this, [=]() {
+            _logStatusMessages();
+
+            if (_updater.isDone()) {
+                emit finished(!_updater.hasError());
+                timer->stop();
+            }
+        });
+
+        _updater.start();
+        timer->start();
+    }
+
+signals:
+    void finished(bool successfully);
+
+private:
+    appimage::update::Updater _updater;
+};
+
+} // namespace
+
+AppImageUpdater::AppImageUpdater(const QUrl &url)
+    : OCUpdater(url)
+{
+}
+
+
+bool AppImageUpdater::handleStartup()
+{
+    // nothing to do, update will be performed while app is running, if anything
+    return false;
+}
+
+void AppImageUpdater::versionInfoArrived(const UpdateInfo &info)
+{
+    const auto &currentVersion = Version::versionWithBuildNumber();
+    const auto newVersion = QVersionNumber::fromString(info.version());
+
+    if (info.version().isEmpty() || currentVersion >= newVersion) {
```

Comment:
I would expect that both evaluate to the same result (no change needed)

Reply:
I tried to stay close to the other updater code, which also performs this check. There, the message "No version information is available at the moment" is logged.